gemac/management.py

```diff
@@ -0,0 +1,155 @@
+        self.wrdata = wrdata[16:]
+
+
+@block
+def management(host_interface, mdio_interface):
+
+    """
+    Management Data - Input Output Interface.
+
+    hostClk - Reference Clk for Management/Configuration Operations. >10MHz
+    hostOpcode (2 Bits) - Define Operation for MDIO Interface. Bit 1 also
+        used as control signal for configuration data transfer.
+    hostAddr (10 Bits) - Address of Register to be accessed. According to
+        Table 8.2, Page 27, Xilinx UG 144 doc.
+    hostWriteData (32 bits) - Data write.
+    hostReadData (32 bits) - Data read.
+    hostMIIM_sel - Set by Host. When High MDIO interface Accessed,
+        Else Configuration registers.
+    hostReq - Set by Host to Indicate ongoing MDIO transaction.
+    hostMIIM_rdy - Set by MDIO Interface to indicate ready for new transaction.
+
+    mdc - Management Clock: programmable frequency derived from host_clk.
+    mdioIn - Input data signal from PHY for its configuration and
+        status.(TriStateBuffer Connected)
+    mdioOut - Output data signal from PHY for its configuration and
+        status.(TriStateBuffer Connected)
+    mdioTri - TriState Control for Signals - Low Indicating mdioOut to be
+        asserted to the MDIO bus.
+
+    """
+
+    mdioout = Signal(bool(0))
+    # mdioin = Signal(bool(0))
+    mdioin = TristateSignal(bool(0))
+    mdioindriver = mdioin.driver()
+    mdiotri = Signal(bool(0))
+    mdio_interface.mdiodriver = mdio_interface.mdioio.driver()
+
+    configregisters = [Signal(intbv(0)[32:0]) for _ in range(10)]
+
+    def getregisternumber(addr):
+        if addr >= 0x200 and addr <= 0x23F:
+            return rx0
+        elif addr >= 0x240 and addr <= 0x27F:
+            return rx1
+        elif addr >= 0x280 and addr <= 0x2BF:
+            return tx
+        elif addr >= 0x2C0 and addr <= 0x2FF:
+            return flow
+        elif addr >= 0x340 and addr <= 0x37F:
+            return managementreg
+        elif addr >= 0x380 and addr <= 0x383:
+            return ucast0
+        elif addr >= 0x384 and addr <= 0x387:
+            return ucast1
+        elif addr >= 0x388 and addr <= 0x38B:
+            return addrtable0
+        elif addr >= 0x38C and addr <= 0x38F:
+            return addrtable1
+        elif addr >= 0x390 and addr <= 0x393:
+            return addrfiltermode
+        else:
+            raise Exception
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def readConfig():
+        if((not host_interface.miimsel) and host_interface.regaddress[9] and
+                host_interface.opcode[1]):
+            host_interface.rddata.next = \
+                configregisters[getregisternumber(host_interface.regaddress)]
+
+    @always_seq(host_interface.clk.posedge, reset=None)
+    def writeConfig():
+        if((not host_interface.miimsel) and host_interface.regaddress[9] and
+                (not host_interface.opcode[1])):
+            configregisters[getregisternumber(host_interface.regaddress)]\
```

Comment:
Made the change. Have to see if its convertible(it would be good as it will reduce redundancy).
host_interface is interface used by host for accessing or changing GEMAC configuration and performing MDIO operations. 
Definition : https://github.com/ravijain056/GEMAC/blob/dev/gemac/interfaces.py